game-core/src/main/java/games/strategy/triplea/delegate/battle/steps/fire/aa/OffensiveAaFire.java

```diff
@@ -0,0 +1,40 @@
+package games.strategy.triplea.delegate.battle.steps.fire.aa;
+
+import games.strategy.engine.data.GamePlayer;
+import games.strategy.engine.data.Unit;
+import games.strategy.engine.delegate.IDelegateBridge;
+import games.strategy.triplea.delegate.ExecutionStack;
+import games.strategy.triplea.delegate.battle.BattleActions;
+import games.strategy.triplea.delegate.battle.BattleState;
+import java.util.Collection;
+
+/** Offensive Aa units can fire and the player can select their casualties */
+public class OffensiveAaFire extends AaFireAndCasualtyStep {
+  private static final long serialVersionUID = 5843852442617511691L;
+
+  public OffensiveAaFire(final BattleState battleState, final BattleActions battleActions) {
+    super(battleState, battleActions);
+  }
+
+  @Override
+  public void execute(final ExecutionStack stack, final IDelegateBridge bridge) {
+    if (valid()) {
```

Comment:
Will we call `execute` of this action if `valid()` were ever false? Asked in another way, do we only call `execute` when `valid()` is true?

Reply:
For steps in general, we will call `execute` when `valid()` is false.  For this specific step, I don't think so.  So I could remove it here.  But I've been trying to give each step the same "structure" to help reduce my cognitive load.

Reply:
If you expect `valid()` to always be true, perhaps better to assert it being true:
```
Preconditions.assertState(valid())
```
This way:
1. It becomes a documented expectation and readers can learn of this assumption
2. If the condition is violated while we are doing any kind of demo/testing, we'll have found a bug